Release notes and upgrading guide for the client libraries (#32989)
keycloak/keycloak: docs/documentation/upgrading/topics/upgrade_client_libs.adoc

= Upgrading the {project_name} Client Libraries

[[_upgrade_client_libraries]]

The client libraries are those artifacts:

* Java admin client - Maven artifact `org.keycloak:keycloak-admin-client`
* Java authorization client - Maven artifact `org.keycloak:keycloak-authz-client`
* Java policy enforcer - Maven artifact `org.keycloak:keycloak-policy-enforcer`

> The client libraries are supported with the last few released server versions. So it is possible that you do not necessarily need to update the server
> at the same time when updating client libraries of your application. This will make the major updates easier as application can be updated independently to
> the {project_name} server major update.

Reviewer: ```suggestion
The client libraries are supported with the last few server versions. So you may not need to update the server
when you update client libraries of your application. Removing this update constraint makes the major updates easier because applications can be updated independently of
the {project_name} server major update.
```

The client library will be always supported with the same major server version as it's own version and with the previous major server version to that one. So for instance client libraries of the 27 stream

> will be supported with the {project_name} server 27 and the {project_name} server 26 major release streams.

Reviewer: @mposolda I hope that my suggestions are correct that it applies to .x releases. So use these suggestions with discretion.

> will be supported with the {project_name} server 27 and the {project_name} server 26 major release streams.

Author: Yes, I think it is correct. Thanks!
